modules/apps/transfer/module.go

```diff
@@ -434,3 +434,17 @@ func (am AppModule) OnTimeoutPacket(
 
 	return nil
 }
+
+// NegotiateAppVersion implements the IBCModule interface
+func (am AppModule) NegotiateAppVersion(
+	ctx sdk.Context,
+	portID string,
+	counterparty channeltypes.Counterparty,
+	counterpartyVersion string,
+) (version string, err error) {
+	if counterpartyVersion != types.Version {
+		return "", sdkerrors.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: got: %s, expected %s", counterpartyVersion, types.Version)
```

Comment:
I think we can safely return an error here if there is a version mismatch.
One would imagine that a hypothetical `"ics20-2"` would involve breaking changes